mne/time_frequency/spectrum.py

```diff
@@ -1179,34 +1214,73 @@ def __getitem__(self, item):
         return BaseRaw._getitem(self, item, return_times=False)
 
 
-def _check_data_shape(data, freqs, info, ndim):
-    if data.ndim != ndim:
-        raise ValueError(f"Data must be a {ndim}D array.")
+def _check_data_shape(
+    data, info, freqs, is_epoched, events, method, standard_ndim, extra_ndim, mt_weights
+):
+    _check_option("data.ndim", data.ndim, (standard_ndim, extra_ndim))
+
+    dims = ()
+    if is_epoched:
+        dims += ("epoch",)
+        if events is not None and data.shape[0] != events.shape[0]:
+            raise ValueError(
+                f"The first dimension of `data` ({data.shape[0]}) must match the first "
+                f"dimension of `events` ({events.shape[0]})."
+            )
```

Comment:
I would move this check to the EpochsSpectrumArray constructor, just after the call to `_check_data_shape`. It's not really shared code since we always have `events=None` for the non-epochs SpectrumArray constructor

Reply:
Done!
https://github.com/mne-tools/mne-python/blob/322ae92f4879d24a31d9ddbd1c504c594dbd1966/mne/time_frequency/spectrum.py#L1601-L1605